components/datatypes/time-profiling.schema.json

```diff
@@ -0,0 +1,37 @@
+{
+  "meta:license": [
+    "Copyright 2020 Adobe Systems Incorporated. All rights reserved.",
+    "This work is licensed under a Creative Commons Attribution 4.0 International (CC BY 4.0) license",
+    "you may not use this file except in compliance with the License. You may obtain a copy",
+    "of the License at https://creativecommons.org/licenses/by/4.0/"
+  ],
+  "$id": "https://ns.adobe.com/xdm/datatypes/time-profiling",
+  "$schema": "http://json-schema.org/draft-06/schema#",
+  "title": "Time Profiling",
+  "type": "object",
+  "meta:extensible": true,
+  "description": "This datatype is used where ever you want to store the time profiling information of any service or component.",
+  "definitions": {
+    "time-profiling": {
+      "properties": {
+        "xdm:startTime": {
+          "title": "startTime",
+          "description": "Execution Start time in Epoch millis for the component.",
+          "type": "number"
```

Comment:
rather storing the epoch millis, should we store this in Date-Time format? Data type will be more explicit in way, and we can use the sql date time methods in queries without any typecasting.
 We can specify type as: 

- "type": "string",
- "format": "date-time"

Reply:
I feel, we should still be using the date-time format for specifying the start/end date. If we keep type as number, we won't get any automatic validation, whether some one has entered the valid date or not. Number means any negative integer also.

Anyway while querying the data we can convert the date-time format to epoch or use sql datediff function for finding the difference between 2 dates.

Also, I see this as a general practice at other places in XDM: https://github.com/adobe/xdm/blob/cf79e096f7f16270b133b0e8609d04b6faf7cb2f/docs/reference/adobe/experience/decisioning/activity-detail.schema.json

Reply:
@sawhneyh Changed the format of startTime and endTime to date-time.